### PR TITLE
US only waitlist

### PR DESCRIPTION
**Pull request by aitorvs**
Task/Issue URL: https://app.asana.com/0/488551667048375/1205785591351675/f

### Description
See asana

### Steps to test this PR
See asana

### CODE REVIEW

**Comment by aitorvs** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #3746** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3746" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/duckduckgo/Android/3746?utm_source=stack-comment).

File: network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/waitlist/NetworkProtectionWaitlistImpl.kt [hunk 1]
```diff
@@ -141,3 +147,36 @@ class NetPRemoteFeatureWrapper @Inject constructor(
...
+@Module
+@ContributesTo(AppScope::class)
+class TimezoneProviderModule {
+    @Provides
+    @InternalApi
+    fun provideUsTimezoneChecker(): USTimeZoneChecker {
+        return {
+            val defaultTimeZone = TimeZone.getDefault()
+            val usTimeZoneIDs = arrayOf(
+                "America/New_York",
```
Comment (karlenDimla) [inline network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/waitlist/NetworkProtectionWaitlistImpl.kt:171]: Any reason why we are not including: Adak, Metlakatla and Phoenix here? I see that they also belong to US timezone.

File: network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/waitlist/NetworkProtectionWaitlistImpl.kt [hunk 2]
```diff
@@ -141,3 +147,36 @@ class NetPRemoteFeatureWrapper @Inject constructor(
...
+                "America/Chicago",
+                "America/Denver",
+                "America/Los_Angeles",
+                "America/Anchorage",
+                "America/Honolulu",
```
Comment (karlenDimla) [inline network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/waitlist/NetworkProtectionWaitlistImpl.kt:176]: I was testing on my Pixel 6 and I get this value when selecting Unites States > Honolulu: `Pacific/Honolulu`

**Review by karlenDimla — APPROVED** [review]
Left a few questions that may/may not need to resolved.
Tested and works as expected on most cases.

File: network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/waitlist/NetworkProtectionWaitlistImpl.kt (hunk 1 above)
Comment (aitorvs) [reply network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/waitlist/NetworkProtectionWaitlistImpl.kt:171]: No, I may be missing more timezones, will include them

Comment (karlenDimla) [reply network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/waitlist/NetworkProtectionWaitlistImpl.kt:175]: Can also check if id starts with “America/*” but we will miss out on Honolulu which is in reality quite far from the US region.

**Pull request merged**